x-pack/plugins/observability_solution/dataset_quality/public/components/dataset_quality_details/degraded_field_flyout/possible_mitigations/manual/index.tsx

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiLink,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  EuiCopy,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+import { MANAGEMENT_APP_ID } from '@kbn/deeplinks-management/constants';
+import { useKibanaContextForPlugin } from '../../../../../utils';
+import {
+  manualMitigationCustomPipelineText,
+  otherMitigationsCustomComponentTemplate,
+  otherMitigationsCustomIngestPipeline,
+} from '../../../../../../common/translations';
+import { useDatasetQualityDetailsState } from '../../../../../hooks';
+import { getComponentTemplatePrefixFromIndexTemplate } from '../../../../../../common/utils/component_template_name';
+
+export function ManualMitigations() {
+  const { integrationDetails } = useDatasetQualityDetailsState();
+  const isIntegration = !!integrationDetails?.integration;
+  return (
+    <>
+      <EditComponentTemplate isIntegration={isIntegration} />
+      <EuiSpacer size="s" />
+      <EditPipeline isIntegration={isIntegration} />
+    </>
+  );
+}
+
+function EditComponentTemplate({ isIntegration }: { isIntegration: boolean }) {
+  const {
+    services: { application },
+  } = useKibanaContextForPlugin();
+
+  const { dataStreamSettings, datasetDetails } = useDatasetQualityDetailsState();
+  const { name } = datasetDetails;
+
+  const onClickHandler = useCallback(async () => {
+    await application.navigateToApp(MANAGEMENT_APP_ID, {
+      path: isIntegration
+        ? `/data/index_management/component_templates/${getComponentTemplatePrefixFromIndexTemplate(
+            dataStreamSettings?.indexTemplate ?? name
+          )}@custom`
+        : `/data/index_management/templates/${dataStreamSettings?.indexTemplate}`,
+      openInNewTab: true,
```

Comment:
**doubt**: don't we have any locator for this yet? Have you considered extending the `indexManagementLocator` to make this easier to maintain and access? It shouldn't be an expensive update given the locator is already setup.

Reply:
Tbh, i didn't dig very deep but at a high level I couldn't find anything explicitly in the management locator for this. I saw other plugins using it exactly like this and hence kept mine implementation similar. 

We can extend the management locator but that would increased the scope of this work alot and hence i didn't explore that part.

Reply:
I looked at the [management plugin locator](https://github.com/elastic/kibana/blob/main/src/plugins/management/common/locator.ts) and changes are minimal to achieve what we need, but would make the access to those pages cleaner and do some work for other teams too. 
Something like this should work to make the locator more flexible. I think it's worth the effort since it implies changing just a couple of files in the PR.
```ts
export type ManagementAppLocatorParams = SerializableRecord &
  (
    | {
        sectionId: string;
        appId?: string;
      }
    | {
        componentTemplate: string;
      }
    | {
        indexTemplate: string;
      }
    | {
        pipeline: string;
      }
  );

export type ManagementAppLocator = LocatorPublic<ManagementAppLocatorParams>;

export class ManagementAppLocatorDefinition
  implements LocatorDefinition<ManagementAppLocatorParams>
{
  public readonly id = MANAGEMENT_APP_LOCATOR;

  public readonly getLocation = async (params: ManagementAppLocatorParams) => {
    const path = buildPathFromParams(params);

    return {
      app: MANAGEMENT_APP_ID,
      path,
      state: {},
    };
  };
}

const buildPathFromParams = (params: ManagementAppLocatorParams) => {
  if (params.sectionId) {
    return `/${params.sectionId}${params.appId ? '/' + params.appId : ''}`;
  }

  if (params.indexTemplate) {
    return `/data/index_management/templates/${params.indexTemplate}`;
  }

  if (params.componentTemplate) {
    return `/data/index_management/component_templates/${params.componentTemplate}`;
  }

  if (params.pipeline) {
    return `/ingest/ingest_pipelines/?pipeline=${params.pipeline}`;
  }

  return '/';
};

```